packages/beacon-node/src/util/blobs.ts

```diff
@@ -46,3 +66,191 @@ export function computeBlobSidecars(
     return {index, blob, kzgCommitment, kzgProof, signedBlockHeader, kzgCommitmentInclusionProof};
   });
 }
+
+/**
+ * Turns a SignedBeaconBlock and an array of Blobs from a given slot into an array of
+ * DataColumnSidecars that are ready to be served by gossip and req/resp.
+ *
+ * Implementation of get_data_column_sidecars
+ * https://github.com/ethereum/consensus-specs/blob/dev/specs/_features/eip7594/das-core.md#get_data_column_sidecars
+ */
+export function computeDataColumnSidecars(
+  config: ChainForkConfig,
+  signedBlock: SignedBeaconBlock,
+  contents: fulu.Contents & {kzgCommitmentsInclusionProof?: fulu.KzgCommitmentsInclusionProof; cells?: fulu.Cell[][]}
+): fulu.DataColumnSidecars {
+  const blobKzgCommitments = (signedBlock as deneb.SignedBeaconBlock).message.body.blobKzgCommitments;
+  if (blobKzgCommitments === undefined) {
+    throw Error("Invalid block with missing blobKzgCommitments for computeBlobSidecars");
+  }
+  if (blobKzgCommitments.length === 0) {
+    return [];
+  }
+  const fork = config.getForkName(signedBlock.message.slot);
+  const signedBlockHeader = signedBlockToSignedHeader(config, signedBlock);
+  const kzgCommitmentsInclusionProof =
+    contents.kzgCommitmentsInclusionProof ?? computeKzgCommitmentsInclusionProof(fork, signedBlock.message.body);
+  const {blobs, kzgProofs} = contents;
+  const cellsAndProofs = Array.from({length: blobs.length}, (_, rowNumber) => {
+    const cells = contents.cells?.[rowNumber] ?? kzg.computeCells(blobs[rowNumber]);
+    const proofs = kzgProofs.slice(rowNumber * NUMBER_OF_COLUMNS, (rowNumber + 1) * NUMBER_OF_COLUMNS);
+    return {cells, proofs};
+  });
+
+  return Array.from({length: NUMBER_OF_COLUMNS}, (_, columnIndex) => {
+    // columnIndex'th column
+    const column = Array.from({length: blobs.length}, (_, rowNumber) => cellsAndProofs[rowNumber].cells[columnIndex]);
+    const kzgProofs = Array.from(
+      {length: blobs.length},
+      (_, rowNumber) => cellsAndProofs[rowNumber].proofs[columnIndex]
+    );
+    return {
+      index: columnIndex,
+      column,
+      kzgCommitments: blobKzgCommitments,
+      kzgProofs,
+      signedBlockHeader,
+      kzgCommitmentsInclusionProof,
+    };
+  });
+}
+
+/**
+ * If the node obtains 50%+ of all the columns, it SHOULD reconstruct the full data matrix via the recover_matrix helper
+ * See https://github.com/ethereum/consensus-specs/blob/dev/specs/fulu/das-core.md#recover_matrix
+ */
+export async function recoverDataColumnSidecars(
```

Comment:
unused function, should be removed?

Reply:
ah whoops. It's used within this file. The confusing thing is there's another `recoverDataColumnSidecars` in src/util/dataColumns.ts which is actually used outside the file.

Reply:
> The confusing thing is there's another recoverDataColumnSidecars

should consider renaming one of them